src/native/libs/System.IO.Compression.Native/CMakeLists.txt

```
@@ -184,7 +184,12 @@ else ()
 endif ()
 
 if((NOT CLR_CMAKE_USE_SYSTEM_ZLIB) AND STATIC_LIBS_ONLY)
-    install_static_library(zlib aotsdk nativeaot)
+    if (CLR_CMAKE_TARGET_UNIX)
+        # zlib on Unix needs to be installed in the same location as System.IO.Compression.Native so that we can then treat is as a 'z' native library.
+        install_static_library(zlib ${STATIC_LIB_DESTINATION} nativeaot)
+    else()
+        install_static_library(zlib aotsdk nativeaot)
+    endif()
```

```suggestion
     install_static_library(zlib ${STATIC_LIB_DESTINATION} nativeaot)
```
Why should be Unix vs. Windows on a different plan?

---

Windows does not use `UseSystemZlib` and did not hit the same problem in the sdk CI.

If I were to change [this line](https://github.com/dotnet/runtime/blob/cf08d43a932e85504e9843a4d6077aace84f5005/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Windows.targets#L49) to use a different folder, then [the EscapePath line below](https://github.com/dotnet/runtime/blob/cf08d43a932e85504e9843a4d6077aace84f5005/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Windows.targets#L60) would not work since it still expects the file to be located in `IlcSdkPath`.

---

That was my first option btw but I didn't want to make heavy changes that would affect many other libraries and instead I opted for making the cmake change proposed here. I'm open to alternatives.

---

Ah, ok. This sounds like a fallout from zlibng being built using different pattern than all other naot static libraries on Windows. We have discussed that before. I guess it is ok then.